phoenix-core/src/main/java/org/apache/phoenix/util/PhoenixRuntime.java

```diff
@@ -459,19 +459,30 @@ public static ColumnInfo getColumnInfo(PTable table, String columnName) throws S
         return getColumnInfo(pColumn);
     }
 
-   /**
+    /**
      * Constructs a column info for the supplied pColumn
      * @param pColumn
      * @return columnInfo
      * @throws SQLException if the parameter is null.
      */
     public static ColumnInfo getColumnInfo(PColumn pColumn) throws SQLException {
-        if (pColumn==null) {
+        if (pColumn == null) {
             throw new SQLException("pColumn must not be null.");
         }
         int sqlType = pColumn.getDataType().getSqlType();
-        ColumnInfo columnInfo = new ColumnInfo(pColumn.toString(),sqlType);
-        return columnInfo;
+        if (pColumn.getMaxLength() == null) {
+            return new ColumnInfo(pColumn.toString(), sqlType);
+        }
+        if (sqlType == Types.CHAR || sqlType == Types.VARCHAR) {
```

Comment:
Rather than check for particular types, it'd be more general to check for null like this:

    Integer maxLength = pColumn.getMaxLength();
    Integer scale = pColumn.getScale();
    return new ColumnInfo(pColumn.toString(), sqlType, maxLength, scale);

Then make sure that ColumnInfo handles a null maxLength and scale.